vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -86,6 +88,25 @@ void WindowsModalHostComponentView::EnsureModalCreated() {
     throw std::exception("Failed to create new hwnd for Modal: " + GetLastError());
   }
 
+  // set the top-level windows as the new hwnd
+  winrt::Microsoft::ReactNative::ReactCoreInjection::SetTopLevelWindowId(host.InstanceSettings().Properties(), reinterpret_cast<uint64_t>(m_hwnd));
+
+  // get current compositor - handles the creation/manipulation of visual objects
+  auto compositionContext = winrt::Microsoft::ReactNative::Composition::implementation::CompositionUIService::GetCompositionContext(m_reactContext.Properties().Handle());
+  auto compositor = winrt::Microsoft::ReactNative::Composition::Experimental::MicrosoftCompositionContextHelper::InnerCompositor(compositionContext);
+
+  // create a react native island - code taken from CompositionHwndHost
+  auto bridge = winrt::Microsoft::UI::Content::DesktopChildSiteBridge::Create(compositor, winrt::Microsoft::UI::GetWindowIdFromWindow(m_hwnd));
+  auto reactNativeIsland = winrt::Microsoft::ReactNative::ReactNativeIsland(compositor);
+  auto contentIsland = reactNativeIsland.Island();
+  bridge.Connect(contentIsland);
+  bridge.Show();
+  bridge.ResizePolicy(winrt::Microsoft::UI::Content::ContentSizePolicy::ResizeContentToParentWindow);
+  m_rootVisual = reactNativeIsland.RootVisual().try_as<winrt::Microsoft::UI::Composition::ContainerVisual>();
+
+  // set ScaleFactor
+  reactNativeIsland.ScaleFactor(GetDpiForWindow(m_hwnd) / 96.0f);
```

Comment:
@acoates-ms UI is loading, thanks so much for the help!!

Another question the UI is not interactable yet, I'm assuming it's because I haven't set the ReactNativeIsland.ReactViewHost(), however should Modal share a ReactViewHost with the main application or create it's own?

Reply:
Hmm, if we dont have a ReactViewHost, then we shouldn't be using ReactNativeIsland. -- At least not in its current form.  ReactNativeIsland is for hosting UI that is a react root view.  We need something similar to host a ReactFragment. 

Not sure what exactly the design for this should look like.
Maybe we create a ReactNativeFragmentIsland class that is similar to ReactNativeIsland, but takes a ComponentView instead of a ReactViewHost.  Or we could try to add such a property to ReactNativeIsland.

The actual reason that the input isn't happening is that no-one is hooking up a CompositionEventHandler to your window, since ReactNativeIsland normally does that when the rootview is initialized.